### DISCUSSION TITLE
Two Websockets were created, but only one could be used at the same time

### DISCUSSION THREAD

**Discussion by minm2015** [fastapi/fastapi#8704] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import time
from fastapi import FastAPI, WebSocket
from service.binance  import tocker_price_service
import uvicorn

app = FastAPI()


@app.websocket("/ws/BTCUSDT")
async def websocket_endpoint_btc(websocket: WebSocket):
    await websocket.accept()
    data = await websocket.receive_text()
    print(data)
    while True:
        time.sleep(1)
        result = await tocker_price_service(['BTCUSDT'])
        await websocket.send_json(f'{result}')

@app.websocket("/ws/ETHUSDT")
async def websocket_endpoint_eth(websocket: WebSocket):
    await websocket.accept()
    data = await websocket.receive_text()
    print(data)
    while True:
        time.sleep(1)
        result = await tocker_price_service(['ETHUSDT'])
        await websocket.send_json(f'{result}')


if __name__ == '__main__':
    uvicorn.run("test:app", host="127.0.0.1", port=8002, reload=True)
```

### Description

When I am using one websocket, the other webSocket is unavailable

Connect a content to return normally
![image](https://user-images.githubusercontent.com/19508192/146031367-6833e085-9eab-4891-b1e2-7d7956baf9be.png)

And the other one has no content
![image](https://user-images.githubusercontent.com/19508192/146031345-9e74854d-b07f-40e9-b778-f391bf7371c6.png)

### Wanted Solution

none

### Wanted Code

```python
nonenonenonenone
```

### Alternatives

none

### Operating System

macOS

### Operating System Details

none

### FastAPI Version

0.70.0

### Python Version

3.9.6

### Additional Context

五

**Comment by jgould22** [comment, marked as answer]
This is why you should not use blocking code in async python.

Use `await asyncio.sleep(1)`, your `sleep` call is blocking the event loop and thus the server cannot do anything while it is sleeping.

**Comment by suryan-s** [comment]
> This is why you should not use blocking code in async python.
> 
> Use `await asyncio.sleep(1)`, your `sleep` call is blocking the event loop and thus the server cannot do anything while it is sleeping.


I have a small doubt. When i tried running the code given by @minm2015 without any time delay or asyncio.sleep, still had the issue of able to run only a single socket connection. does this have to do anything with the actual working of sockets or asynchronous programming? Is giving a delay mandatory?
